src/lex.js

```diff
@@ -1390,7 +1407,7 @@ Lexer.prototype = {
     // Check regular expression for correctness.
 
     try {
-      new RegExp(body, flags.join(""));
+      new RegExp(body);
```

Comment:
This fails for e.g. `/\u{1}/u` and `/[💩-💫]/u`.

See https://mathiasbynens.be/notes/es6-unicode-regex.